src/pages/api/sanity/lessons/create.ts

```diff
@@ -180,7 +188,8 @@ const createSanityLessons = async (
     } else {
       let transaction = sanityClient.transaction()
 
-      const {course, lessons} = req.body
+      const course = courseSchema.parse(req.body.course)
+      const lessons = z.array(lessonSchema).parse(req.body.lessons)
 
       const {sanityCourse, sanityLessons, sanityResources} =
         await formatSanityMutationForLessons(course, lessons)
```

Comment:
Validate incoming request data before passing it along to the mutation formatter. If anything is off, it will throw. If extra, unexpected data is coming in, that will be stripped out by the `parse`. Clean!